src/server/helpers/achievement.js

```diff
@@ -474,71 +454,72 @@ function achievementToUnlockId(achievementUnlock) {
  * @param {int} revisionId - Revision to get release date of
  * @returns {int} - Days since edition was released
  */
-function getEditionDateDifference(orm, revisionId) {
+async function getEditionDateDifference(orm, revisionId) {
 	const {EditionRevision} = orm;
-	return new EditionRevision({id: revisionId}).fetch()
-		.then((edition) => edition.related('data').fetch())
-		.then((data) => data.related('releaseEventSet').fetch())
-		.then(
-			(releaseEventSet) =>
-				releaseEventSet.related('releaseEvents').fetch()
-		)
-		.then((releaseEvents) => {
-			let differencePromise;
-			if (releaseEvents.length > 0) {
-				const msInOneDay = 86400000;
-				const attribs = releaseEvents.models[0].attributes;
-				const date = new Date(
-					attribs.year,
-					attribs.month - 1,
-					attribs.day
-				);
-				const now = new Date(Date.now());
-				differencePromise =
-					Math.round((date.getTime() - now.getTime()) / msInOneDay);
-			}
-			else {
-				differencePromise =
-				new Promise((resolve, reject) => reject(new Error('no date attribute')));
-			}
-			return differencePromise;
-		})
-		.catch(() => new Promise((resolve, reject) => reject(new Error('no date attribute'))));
+	try {
+		const edition = await new EditionRevision({id: revisionId}).fetch();
+		const data = await edition.related('data').fetch();
+		const releaseEventSet = await data.related('releaseEventSet').fetch();
+		const releaseEvents = await releaseEventSet.related('releaseEvents').fetch();
+		let differencePromise;
+		if (releaseEvents.length > 0) {
+			const msInOneDay = 86400000;
+			const attribs = releaseEvents.models[0].attributes;
+			const date = new Date(
+				attribs.year,
+				attribs.month - 1,
+				attribs.day
+			);
+			const now = new Date(Date.now());
+			differencePromise = Math.round((date.getTime() - now.getTime()) / msInOneDay);
```

Comment:
While we're in there touching the code, this can be improved:
```suggestion
			const now = Date.now();
			return Math.round((date.getTime() - now) / msInOneDay);
```